core/core.go

```diff
@@ -382,31 +382,76 @@ func makeAddrsFactory(cfg config.Addresses) (p2pbhost.AddrsFactory, error) {
 	}, nil
 }
 
-func makeSmuxTransport(mplexExp bool) smux.Transport {
+func makeSmuxTransport(multiplexers []string, mplexExp bool) smux.Transport {
 	mstpt := mssmux.NewBlankTransport()
 
-	ymxtpt := &yamux.Transport{
-		AcceptBacklog:          512,
-		ConnectionWriteTimeout: time.Second * 10,
-		KeepAliveInterval:      time.Second * 30,
-		EnableKeepAlive:        true,
-		MaxStreamWindowSize:    uint32(1024 * 512),
-		LogOutput:              ioutil.Discard,
+	multiplexerProtocolNames := map[string]string{
+		"yamux": "/yamux/1.0.0",
+		"mplex": "/mplex/6.7.0",
 	}
 
-	if os.Getenv("YAMUX_DEBUG") != "" {
-		ymxtpt.LogOutput = os.Stderr
+	var isYamux, isMplex bool
+
+	for _, multiplexer := range multiplexers {
+		if multiplexer == "yamux" {
+			isYamux = true
+		} else if multiplexer == "mplex" {
```

Comment:
@Stebalien @whyrusleeping do you think using just protocol names is good enough here or should we use full protocol descriptor. This way if we ever want to have mplex6 and mplex7 (if it ever exists) both at the same time, it will be possible.

Reply:
I don't have much of a preference.

Reply:
@Stebalien @Kubuxu sorry for letting this get stale. i can definitely update the pr to use full protocol descriptors if you guys think that's better.

Reply:
I don't think this is necessary. We can just have `mplex` and in future add others.